### ISSUE TITLE
Any plans to release EfficientNet based SSD for object detection ? [Question]

### ISSUE THREAD

**Issue by dexception** [tensorflow/models#6996]
Any plans to release EfficientNet based SSD for object detection ?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by dexception** [comment]
What is the top-level directory of the model you are using : N/A
Have I written custom code: No
OS Platform and Distribution: Ubuntu 16.04
TensorFlow installed from: Pip
TensorFlow version: 1.13.1
Bazel version: N/A
CUDA/cuDNN version: 9.0
GPU model and memory: 1080 TI
Exact command to reproduce: N/A

**Comment by lukasschmit** [comment]
I'm also very interested in this. I'm currently using the tpu implementations of [retinanet](https://github.com/tensorflow/tpu/tree/master/models/official/retinanet) and [efficientnet](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet) together, but that code is very centered on tpu training and I've had to rewrite a bunch just to make it work on GPUs. Plus, the TFRecord input format is a pain file-size-wise. Would be great to get an implementation with better support for people who can't afford TPUs.

**Comment by PythonImageDeveloper** [comment]
Hi @lukasschmit 
How do you train the model on tpu device? Do you use the google colab service? When the model trained using tpu device, Can extract the freeze_graph file (.pb) using the GPU device or necessary only tpu device and tpu support python file?

**Comment by dexception** [comment]
This looks like a complete unprofessional project.

No Imagenet models for the following:

1. ResneXt
2. EfficientNet B0-B7
3. ShuffleNet
4. Mixnet S-M-L
5. MobileNetV3
6. FbNet
7. SENet 
8. GhostNet : https://arxiv.org/abs/1911.11907

**Comment by noahtren** [comment]
There are TF2 (Keras) compatible ImageNet models for EfficientNet, just not part of this repo yet. See https://github.com/keras-team/keras-applications

Regarding EfficientNet for object detection, you might be interested in EfficientDet which is a recent variation of EfficientNet for object detection. This is a very recent paper and the code isn't public yet. https://arxiv.org/abs/1911.09070

There's an issue on tensorflow/tpu tracking when it will be released https://github.com/tensorflow/tpu/issues/611

**Comment by dexception** [comment]
I have to ask... why there has been absolutely no work in this repo in last few months ?

**Comment by dynamicwebpaige** [comment]
A mobile-compatible EfficientNet was [released this month](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet). Please try it out, and make sure to check the [release notes](https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/lite/README.md) for specific features. Thanks!

**Comment by bhack** [comment]
An anchor free version with EfficientDet backbone is available at https://github.com/xuannianz/SAPD.

For others anchor free models https://github.com/tensorflow/hub/issues/424

@dynamicwebpaige  P.s. Users still need to always spread ticket everywhere https://github.com/tensorflow/tpu/issues/611. I hope that we could centralize models.

**Comment by bhack** [comment]
See also https://github.com/tensorflow/tpu/issues/696
https://github.com/tensorflow/tpu/issues/664

**Comment by bhack** [comment]
And I suppose we need also an EfficientDet-lite version after https://blog.tensorflow.org/2020/03/higher-accuracy-on-vision-models-with-efficientnet-lite.html

**Comment by bhack** [comment]
The release was announced today but in another Google (AutoML) repo.. just to have a little bit more fragmentation. 
See https://github.com/tensorflow/tpu/issues/611#issuecomment-600192115

**Comment by saberkun** [comment]
Hi, EfficientDet is open sourced by Mingxing in automl dedicated repository.
Official code which is TF1 implement is here: https://github.com/google/automl/tree/master/efficientdet

We are very close to release EfficientNet image classification training scripts (currently for basic EfficientNet architecture) in tensorflow/models/official/vision.

**Comment by bhack** [comment]
@saberkun I hope that from a pratical point of view we could use It also with new Efficientnet light backbone. /cc @mingxingtan 
I think that having some example in TFHUB with anchor-free solutions It could simplify the pipeline for many in object keypoits/landmarks task (face landmarks, pose keypoints, in object landmarks, etc..).
